### PR TITLE
[Enhancement] fix sonar problems

### PR DESCRIPTION
**Pull request by packy92**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Mainly fix four types problem. 
- rename `finalize` method name to `finish` to avoid any possible confusion with Object.finalize().
- filling the missing `hashCode` or `equals` method. 
- replacing break in loop statement with getting the first element in list/map then do the original work.
- use try with statement to ensure closing resource
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 51 / 120 (42.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndexMeta.java|0|1|00.00%|[157]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|1|00.00%|[699]|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|4|00.00%|[111, 112, 137, 138]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|0|1|00.00%|[435]|
|:large_blue_circle:|com/starrocks/load/FailMsg.java|0|1|00.00%|[94]|
|:large_blue_circle:|com/starrocks/metric/MetricLabel.java|0|1|00.00%|[47]|
|:large_blue_circle:|com/starrocks/catalog/Index.java|0|7|00.00%|[116, 117, 120, 121, 124, 125, 126]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|1|00.00%|[114]|
|:large_blue_circle:|com/starrocks/catalog/MapType.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/load/ExportFailMsg.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/persist/DropInfo.java|0|1|00.00%|[101]|
|:large_blue_circle:|com/starrocks/analysis/FunctionParams.java|0|6|00.00%|[127, 128, 131, 132, 135, 136]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|1|00.00%|[402]|
|:large_blue_circle:|com/starrocks/catalog/RandomDistributionInfo.java|0|1|00.00%|[91]|
|:large_blue_circle:|com/starrocks/persist/BatchModifyPartitionsInfo.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadTaskInfo.java|0|1|00.00%|[232]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|1|00.00%|[201]|
|:large_blue_circle:|com/starrocks/persist/PartitionPersistInfo.java|0|1|00.00%|[136]|
|:large_blue_circle:|com/starrocks/persist/DropPartitionInfo.java|0|1|00.00%|[105]|
|:large_blue_circle:|com/starrocks/catalog/ColumnStats.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/persist/DropDbInfo.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/mysql/MysqlCapability.java|0|1|00.00%|[155]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/JoinOrder.java|0|14|00.00%|[69, 70, 72, 73, 76, 77, 78, 79, 102, 103, 105, 106, 109, 110]|
|:large_blue_circle:|com/starrocks/persist/TablePropertyInfo.java|0|1|00.00%|[110]|
|:large_blue_circle:|com/starrocks/system/Backend.java|0|1|00.00%|[358]|
|:large_blue_circle:|com/starrocks/persist/ColocatePersistInfo.java|0|1|00.00%|[129]|
|:large_blue_circle:|com/starrocks/persist/ModifyPartitionInfo.java|0|1|00.00%|[97]|
|:large_blue_circle:|com/starrocks/persist/ReplicaPersistInfo.java|0|3|00.00%|[355, 356, 357]|
|:large_blue_circle:|com/starrocks/catalog/DataProperty.java|0|1|00.00%|[86]|
|:large_blue_circle:|com/starrocks/catalog/Function.java|0|1|00.00%|[744]|
|:large_blue_circle:|com/starrocks/analysis/BetweenPredicate.java|3|6|50.00%|[79, 85, 87]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|1|2|50.00%|[260]|
|:large_blue_circle:|com/starrocks/analysis/ExistsPredicate.java|3|6|50.00%|[85, 92, 94]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|3|4|75.00%|[406]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|5|6|83.33%|[314]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructField.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DecimalLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/StringLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LiteralExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/IsNullPredicate.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalFilterOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Subquery.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/NullLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/BoolLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/FloatLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/IntLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Replica.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LargeIntLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CaseExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/InPredicate.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalFilterOperator.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 51 / 131 (38.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndexMeta.java|0|1|00.00%|[157]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|1|00.00%|[699]|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|4|00.00%|[111, 112, 137, 138]|
|:large_blue_circle:|com/starrocks/plugin/DynamicPluginLoader.java|0|1|00.00%|[208]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|0|1|00.00%|[435]|
|:large_blue_circle:|com/starrocks/load/FailMsg.java|0|1|00.00%|[94]|
|:large_blue_circle:|com/starrocks/metric/MetricLabel.java|0|1|00.00%|[47]|
|:large_blue_circle:|com/starrocks/catalog/Index.java|0|7|00.00%|[116, 117, 120, 121, 124, 125, 126]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|1|00.00%|[114]|
|:large_blue_circle:|com/starrocks/catalog/MapType.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/load/ExportFailMsg.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/persist/DropInfo.java|0|1|00.00%|[101]|
|:large_blue_circle:|com/starrocks/analysis/FunctionParams.java|0|6|00.00%|[127, 128, 131, 132, 135, 136]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[28, 29, 30, 31]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|1|00.00%|[402]|
|:large_blue_circle:|com/starrocks/catalog/RandomDistributionInfo.java|0|1|00.00%|[91]|
|:large_blue_circle:|com/starrocks/persist/BatchModifyPartitionsInfo.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadTaskInfo.java|0|1|00.00%|[232]|
|:large_blue_circle:|com/starrocks/common/util/BrokerUtil.java|0|1|00.00%|[298]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|1|00.00%|[200]|
|:large_blue_circle:|com/starrocks/persist/PartitionPersistInfo.java|0|1|00.00%|[136]|
|:large_blue_circle:|com/starrocks/persist/DropPartitionInfo.java|0|1|00.00%|[105]|
|:large_blue_circle:|com/starrocks/catalog/ColumnStats.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/persist/DropDbInfo.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/common/util/SmallFileMgr.java|0|5|00.00%|[436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/mysql/MysqlCapability.java|0|1|00.00%|[155]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/JoinOrder.java|0|14|00.00%|[69, 70, 72, 73, 76, 77, 78, 79, 102, 103, 105, 106, 109, 110]|
|:large_blue_circle:|com/starrocks/persist/TablePropertyInfo.java|0|1|00.00%|[110]|
|:large_blue_circle:|com/starrocks/system/Backend.java|0|1|00.00%|[358]|
|:large_blue_circle:|com/starrocks/persist/ColocatePersistInfo.java|0|1|00.00%|[129]|
|:large_blue_circle:|com/starrocks/persist/ModifyPartitionInfo.java|0|1|00.00%|[97]|
|:large_blue_circle:|com/starrocks/persist/ReplicaPersistInfo.java|0|3|00.00%|[355, 356, 357]|
|:large_blue_circle:|com/starrocks/catalog/DataProperty.java|0|1|00.00%|[86]|
|:large_blue_circle:|com/starrocks/catalog/Function.java|0|1|00.00%|[744]|
|:large_blue_circle:|com/starrocks/analysis/BetweenPredicate.java|3|6|50.00%|[79, 85, 87]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|1|2|50.00%|[260]|
|:large_blue_circle:|com/starrocks/analysis/ExistsPredicate.java|3|6|50.00%|[85, 92, 94]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|3|4|75.00%|[406]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|5|6|83.33%|[314]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructField.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DecimalLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/StringLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LiteralExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/IsNullPredicate.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalFilterOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Subquery.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/NullLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/BoolLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/FloatLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/IntLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Replica.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LargeIntLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CaseExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/InPredicate.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalFilterOperator.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 50 / 129 (38.76%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndexMeta.java|0|1|00.00%|[156]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|1|00.00%|[698]|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|4|00.00%|[111, 112, 137, 138]|
|:large_blue_circle:|com/starrocks/plugin/DynamicPluginLoader.java|0|1|00.00%|[208]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|0|1|00.00%|[435]|
|:large_blue_circle:|com/starrocks/load/FailMsg.java|0|1|00.00%|[94]|
|:large_blue_circle:|com/starrocks/metric/MetricLabel.java|0|1|00.00%|[47]|
|:large_blue_circle:|com/starrocks/catalog/Index.java|0|7|00.00%|[116, 117, 120, 121, 124, 125, 126]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|1|00.00%|[114]|
|:large_blue_circle:|com/starrocks/catalog/MapType.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/load/ExportFailMsg.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/persist/DropInfo.java|0|1|00.00%|[101]|
|:large_blue_circle:|com/starrocks/analysis/FunctionParams.java|0|6|00.00%|[127, 128, 131, 132, 135, 136]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[28, 29, 30, 31]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndex.java|0|1|00.00%|[301]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|1|00.00%|[401]|
|:large_blue_circle:|com/starrocks/catalog/RandomDistributionInfo.java|0|1|00.00%|[91]|
|:large_blue_circle:|com/starrocks/persist/BatchModifyPartitionsInfo.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadTaskInfo.java|0|1|00.00%|[232]|
|:large_blue_circle:|com/starrocks/common/util/BrokerUtil.java|0|1|00.00%|[298]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|1|00.00%|[200]|
|:large_blue_circle:|com/starrocks/persist/PartitionPersistInfo.java|0|1|00.00%|[136]|
|:large_blue_circle:|com/starrocks/persist/DropPartitionInfo.java|0|1|00.00%|[105]|
|:large_blue_circle:|com/starrocks/catalog/ColumnStats.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/persist/DropDbInfo.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/common/util/SmallFileMgr.java|0|5|00.00%|[436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/mysql/MysqlCapability.java|0|1|00.00%|[155]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/JoinOrder.java|0|14|00.00%|[69, 70, 72, 73, 76, 77, 78, 79, 102, 103, 105, 106, 109, 110]|
|:large_blue_circle:|com/starrocks/persist/TablePropertyInfo.java|0|1|00.00%|[110]|
|:large_blue_circle:|com/starrocks/system/Backend.java|0|1|00.00%|[358]|
|:large_blue_circle:|com/starrocks/persist/ColocatePersistInfo.java|0|1|00.00%|[129]|
|:large_blue_circle:|com/starrocks/persist/ModifyPartitionInfo.java|0|1|00.00%|[97]|
|:large_blue_circle:|com/starrocks/persist/ReplicaPersistInfo.java|0|1|00.00%|[355]|
|:large_blue_circle:|com/starrocks/catalog/DataProperty.java|0|1|00.00%|[86]|
|:large_blue_circle:|com/starrocks/catalog/Function.java|0|1|00.00%|[744]|
|:large_blue_circle:|com/starrocks/analysis/BetweenPredicate.java|3|6|50.00%|[79, 85, 87]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|1|2|50.00%|[260]|
|:large_blue_circle:|com/starrocks/analysis/ExistsPredicate.java|3|6|50.00%|[85, 92, 94]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|3|4|75.00%|[406]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|5|6|83.33%|[314]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructField.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DecimalLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/StringLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LiteralExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/IsNullPredicate.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalFilterOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Subquery.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/NullLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/BoolLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/FloatLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/IntLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Replica.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LargeIntLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CaseExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/InPredicate.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalFilterOperator.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Group.java [hunk 1]
```diff
@@ -304,6 +304,20 @@ public int hashCode() {
         return id;
     }
 
+    @Override
+    public boolean equals(Object obj) {
```
[the remark this answers is not in the document]
Comment (packy92) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Group.java:308]: The hashcode method ensures different group has different hashcode value. Hence, the equals method doesn't affect the performance when add a group to a set/map. So, can we remain this method if we want compare two group object directly in the future?

**Review by Seaven — DISMISSED** [review]

**Review by HangyuanLiu — DISMISSED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 48 / 127 (37.80%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndexMeta.java|0|1|00.00%|[156]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|1|00.00%|[698]|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|4|00.00%|[111, 112, 137, 138]|
|:large_blue_circle:|com/starrocks/plugin/DynamicPluginLoader.java|0|1|00.00%|[208]|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|0|1|00.00%|[435]|
|:large_blue_circle:|com/starrocks/load/FailMsg.java|0|1|00.00%|[94]|
|:large_blue_circle:|com/starrocks/metric/MetricLabel.java|0|1|00.00%|[47]|
|:large_blue_circle:|com/starrocks/catalog/Index.java|0|7|00.00%|[116, 117, 120, 121, 124, 125, 126]|
|:large_blue_circle:|com/starrocks/catalog/HashDistributionInfo.java|0|1|00.00%|[114]|
|:large_blue_circle:|com/starrocks/catalog/MapType.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/load/ExportFailMsg.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/persist/DropInfo.java|0|1|00.00%|[101]|
|:large_blue_circle:|com/starrocks/analysis/FunctionParams.java|0|6|00.00%|[127, 128, 131, 132, 135, 136]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[28, 29, 30, 31]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndex.java|0|1|00.00%|[301]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|1|00.00%|[401]|
|:large_blue_circle:|com/starrocks/catalog/RandomDistributionInfo.java|0|1|00.00%|[91]|
|:large_blue_circle:|com/starrocks/persist/BatchModifyPartitionsInfo.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadTaskInfo.java|0|1|00.00%|[232]|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|1|00.00%|[865]|
|:large_blue_circle:|com/starrocks/common/util/BrokerUtil.java|0|1|00.00%|[300]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|1|00.00%|[200]|
|:large_blue_circle:|com/starrocks/persist/PartitionPersistInfo.java|0|1|00.00%|[136]|
|:large_blue_circle:|com/starrocks/persist/DropPartitionInfo.java|0|1|00.00%|[105]|
|:large_blue_circle:|com/starrocks/catalog/ColumnStats.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/persist/DropDbInfo.java|0|1|00.00%|[87]|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/common/util/SmallFileMgr.java|0|5|00.00%|[436, 437, 438, 439, 440]|
|:large_blue_circle:|com/starrocks/mysql/MysqlCapability.java|0|1|00.00%|[155]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/JoinOrder.java|0|14|00.00%|[69, 70, 72, 73, 76, 77, 78, 79, 102, 103, 105, 106, 109, 110]|
|:large_blue_circle:|com/starrocks/persist/TablePropertyInfo.java|0|1|00.00%|[110]|
|:large_blue_circle:|com/starrocks/system/Backend.java|0|1|00.00%|[364]|
|:large_blue_circle:|com/starrocks/persist/ColocatePersistInfo.java|0|1|00.00%|[129]|
|:large_blue_circle:|com/starrocks/persist/ModifyPartitionInfo.java|0|1|00.00%|[97]|
|:large_blue_circle:|com/starrocks/load/EtlStatus.java|0|1|00.00%|[260]|
|:large_blue_circle:|com/starrocks/persist/ReplicaPersistInfo.java|0|1|00.00%|[355]|
|:large_blue_circle:|com/starrocks/catalog/DataProperty.java|0|1|00.00%|[125]|
|:large_blue_circle:|com/starrocks/analysis/BetweenPredicate.java|3|6|50.00%|[79, 85, 87]|
|:large_blue_circle:|com/starrocks/analysis/ExistsPredicate.java|3|6|50.00%|[85, 92, 94]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|3|4|75.00%|[406]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Group.java|5|6|83.33%|[314]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructField.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DecimalLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/StringLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LiteralExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/IsNullPredicate.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalFilterOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Subquery.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/NullLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/BoolLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/FloatLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/IntLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Replica.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Function.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/LargeIntLiteral.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/CaseExpr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/InPredicate.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalFilterOperator.java|1|1|100.00%|[]|

**Pull request merged**